### PR TITLE
Feature/#305 - Integrating Mastodon links

### PR DESCRIPTION
**Pull request by FDelporte**
Waiting for a new version of data library cause new method was added:
`person.getMastodonUrl()`

See
https://github.com/dlsc-software-consulting-gmbh/jfxcentral-data/commit/3966489cec7681e13ef7383ba0e78ada9c691e84

### CODE REVIEW

**Comment by dlemmermann** [comment]
I think we also need the mastodon attribute on Person.java. See new issue: https://github.com/dlsc-software-consulting-gmbh/jfxcentral-data/issues/258

**Comment by FDelporte** [comment]
Integration is OK and tested, but needs and extra icon to be added, see
https://github.com/dlemmermann/jfxcentral2/issues/310

**Review by dlemmermann — COMMENTED** [review]
There was no ticket assigned to you for this work. We follow strict "first an issue, then a pull request, a review, a merge" process.

**Comment by dlemmermann** [comment]
@leewyatt please review this pull request

**Comment by dlemmermann** [comment]
And you are right ... man, not my day today. Mondays ....

**Comment by dlemmermann** [comment]
Also, I think I just don't have you on the radar to do frontend stuff for this project :-)

**Comment by FDelporte** [comment]
> Also, I think I just don't have you on the radar to do frontend stuff for this project :-)

Big mistake ;-)

File: app/src/main/java/com/dlsc/jfxcentral2/app/pages/LinksOfTheWeekPage.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ protected List<MenuView.Item> createMenuItems() {
                 .stream()
                 .sorted(Comparator.comparing(LinksOfTheWeek::getCreatedOn).reversed())
                 .map(links -> new MenuView.Item(DATE_FORMATTER.format(links.getCreatedOn()), null, null))
-                .limit(5)
+                .limit(20)
```
Comment (dlemmermann) [inline app/src/main/java/com/dlsc/jfxcentral2/app/pages/LinksOfTheWeekPage.java:96]: That is overkill I think but we can try for now. Also this is a UI change. Please run this by me first.

File: iconfont/src/main/java/com/dlsc/jfxcentral2/iconfont/JFXCentralIcon.java [hunk 2]
```diff
@@ -38,6 +38,7 @@ public enum JFXCentralIcon implements Ikon {
     LINKS("jfx-icon-links", '\ue91f'),
     LOG_IN("jfx-icon-log-in", '\ue920'),
     LOG_OUT("jfx-icon-log-out", '\ue921'),
+    MASTODON("jfx-icon-mastodon", '\ue931'),
```
[the remark this answers is not in the document]
Comment (dlemmermann) [reply iconfont/src/main/java/com/dlsc/jfxcentral2/iconfont/JFXCentralIcon.java:41]: Actually the SVG folder is not a resource folder. It is directly under the module folder for the "iconfont" module. Initially the icon files were sorted alphabetically. But later on we added more icons and we have to keep the order of the files (hence prefixing with a counter). If we do not keep the order then the unicodes will constantly change whenever we rebuild the icon font.

File: components/src/main/java/com/dlsc/jfxcentral2/components/SocialLinksView.java [hunk 3]
```diff
@@ -82,25 +90,28 @@ public SocialLinksView() {
     private void updateView() {
         getChildren().clear();
 
-        if (getTwitterUrl() != null) {
+        if (getTwitterUrl() != null && !getTwitterUrl().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (FDelporte) [reply components/src/main/java/com/dlsc/jfxcentral2/components/SocialLinksView.java]: Fixed

File: iconfont/src/main/java/com/dlsc/jfxcentral2/iconfont/JFXCentralIcon.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (FDelporte) [reply iconfont/src/main/java/com/dlsc/jfxcentral2/iconfont/JFXCentralIcon.java]: Fixed + added TODO to include the correct Mastodon icon code

File: iconfont/src/main/java/com/dlsc/jfxcentral2/iconfont/JFXCentralIcon.java [hunk 4]
```diff
@@ -60,7 +59,10 @@ public enum JFXCentralIcon implements Ikon {
     YOUTUBE("jfx-icon-youtube", '\ue934'),
     TOP_CONTENT("jfx-icon-top-content", '\ue935'),
     HANDSHAKE("jfx-icon-handshake", '\ue936'),
-    TEAM("jfx-icon-team", '\ue937');
+    TEAM("jfx-icon-team", '\ue937'),
+    // TODO Replace Mastodon with correct icon code, current identical to Twitter
+    // See https://github.com/dlemmermann/jfxcentral2/issues/310
+    MASTODON("jfx-icon-mastodon", '\ue931') ;
```
Comment (dlemmermann) [inline iconfont/src/main/java/com/dlsc/jfxcentral2/iconfont/JFXCentralIcon.java:65]: We will not add a new icon in our icon font. Instead use the font and icon suggested by @leewyatt

**Review by dlemmermann — CHANGES_REQUESTED** [review]

File: iconfont/src/main/java/com/dlsc/jfxcentral2/iconfont/JFXCentralIcon.java (hunk 4 above)
Comment (leewyatt) [reply iconfont/src/main/java/com/dlsc/jfxcentral2/iconfont/JFXCentralIcon.java]: ok~

**Review by dlemmermann — APPROVED** [review]

**Pull request merged**